### DISCUSSION TITLE
How to fine tune existing openvino_notebooks model with custom dataset ? for Automatic Self-checkout

### DISCUSSION THREAD

**Discussion by Abhijeet241093** [openvinotoolkit/openvino_notebooks#1802] [AI Solution Kits]
**Describe the bug**
How to fine tune existing open-vino_notebooks model with custom dataset ?

 **Expected behavior**
Any Step or procedure or video recommended to follow it. 

**Screenshots**

![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/38974232/fb1d4353-1f85-4d3a-87a1-a94cadba9323)

![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/38974232/d1a9d390-3b28-45b3-9be8-adf2da2586a1)

![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/38974232/78c5f5bc-e695-4c08-a029-5fbfb470277e)

![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/38974232/56b8a86d-10f5-45e6-bd42-8901c9479a05)

 **Installation instructions (Please mark the checkbox)**
[Yes ] I followed the installation guide at https://github.com/openvinotoolkit/openvino_notebooks#-installation-guide to install the notebooks. 

** Environment information **
Please run `python check_install.py` in the _openvino_notebooks_ directory. If the output is _NOT OK_ for any of the checks, please follow the instructions to fix that. If that does not work, or if you still encounter the issue, please paste the output of check_install.py here.
 
**Additional context**

**Comment by adrianboguszewski** [comment]
@riacheruvu please look at it

**Comment by riacheruvu** [comment]
Hi @Abhijeet241093, this is a great question, thanks! We have a end-to-end tutorial showing how to train a YOLOv8 model with a custom dataset and then export it to the OpenVINO IR format for memory-efficiency here: https://openvinotoolkit.github.io/datumaro/latest/docs/jupyter_notebook_examples/notebooks/08_e2e_example_yolo_ultralytics_trainer.html#Prerequisite. This example uses Ultralytics YOLOv8 trainer approach - you could train a model on your own custom dataset and then use it as part of the automated self-checkout kit notebook. I hope this helps!

**Comment by Abhijeet241093** [comment]
Hi @riacheruvu, thank you very much. Its working fine.

**Comment by Abhijeet241093** [comment]
Hi @riacheruvu,    thank you very much for your attention. Please find error below

**Describe the bug**
While custom training using command :
yolo detect train model=yolov8n.pt data=/home/tricubics/Desktop/Automatic_Self_Checkout/automated_self_checkout/self_checkout_ultralytics/data.yaml epochs=100 imgsz=640 project=self_checkout_project device='0,1'

Getting Message 
Transferred 319/355 items from pretrained weights
DDP: debug command /usr/bin/python3 -m torch.distributed.run --nproc_per_node 2 --master_port 48483 /home/tricubics/.config/Ultralytics/DDP/_temp_99pl7w9c139968793272672.py
WARNING:__main__:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
Ultralytics YOLOv8.1.23 🚀 Python-3.8.10 torch-2.2.1+cu121 CUDA:0 (NVIDIA GeForce RTX 3080, 10001MiB)
                                                           CUDA:1 (NVIDIA GeForce RTX 3080, 10009MiB)
[W Utils.hpp:164] Warning: Environment variable NCCL_BLOCKING_WAIT is deprecated; use TORCH_NCCL_BLOCKING_WAIT instead (function getCvarBool)
[W Utils.hpp:164] Warning: Environment variable NCCL_BLOCKING_WAIT is deprecated; use TORCH_NCCL_BLOCKING_WAIT instead (function getCvarBool)
TensorBoard: Start with 'tensorboard --logdir self_checkout_project/train12', view at http://localhost:6006/
Overriding model.yaml nc=80 with nc=6
Transferred 319/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
**AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...**
Unable to open : http://localhost:6006/

 **Expected behavior**

Program Must run 

**Screenshots**
If applicable, add screenshots to help explain your problem.
![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/38974232/80f602d9-be07-4f6f-9736-1933c375a98c)

**Comment by riacheruvu** [comment]
Hi @Abhijeet241093, answered your question here: #1786; briefly, this is not an issue corresponding to the OpenVINO toolkit or notebooks and I see you have raised the issue on the Ultralytics Github repo, which is the right place.

I will ask @adrianboguszewski to close this thread, but please feel free to create a new [discussion thread](https://github.com/openvinotoolkit/openvino_notebooks/discussions) if you need further support, or an issue if you find a bug with the export process to OpenVINO. Thank you!

**Reply by adrianboguszewski** [reply]
Converted to discussion
